src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -124,6 +128,50 @@ public User addUser(PrimaryKey id) {
 		return productRepository.findTopNRating(PageRequest.of(page, nums));
 	}
 	
+	public ResponseEntity<?> getTopUsers(String selected, int page, int nums) {
+		List<User> users = new ArrayList<>();
+		if(selected.equals("checkout")) {
+			users = userRepository.findUserCheckouts(true);
+		}
+		else if(selected.equals("faves")) {
+			users = userRepository.findUserFavourites(true);
+		}
+		else if(selected.equals("ratings")){
+			users = userRepository.findTopRaters();
+		}
+		
+		LinkedHashMap<Integer, Integer> uniqueId = new LinkedHashMap<>();
+		for(User user: users) {
+			if(!uniqueId.containsKey(user.getId().getUserId())) {
+				uniqueId.put(user.getId().getUserId(), 1);
+			} else {
+				uniqueId.put(user.getId().getUserId(), uniqueId.get(user.getId().getUserId()) + 1);
+			}
+		}
+		System.out.println("hello " + users);
+		List<Entry<Integer,Integer>> result = new ArrayList<>(uniqueId.entrySet());
+		result.sort(Entry.comparingByValue());
+		Collections.reverse(result);
+		if(page*nums-1<=result.size()) {
```

Comment:
This wouldn't work if page = 0. 
In that case, the result populated [here ](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/pull/8/files#diff-0fae860ae0c1e35be827d26bc884c318R156) will most likely be an empty list

Whereas Page =0 is a valid value. Maybe have a special logic that handles page=0